app/assets/javascripts/showTree.js

```diff
@@ -1,401 +1,400 @@
+      });
+
+      new loadChart(d.load, d.name, d.capacity).render('.load-graph svg')
+    };
+
+    function toggleSelectedNode(){
+      d3.selectAll("text").style("font-weight", "normal");
+      d3.selectAll("text").style("text-decoration", "none");
+      d3.select(this).select("text").style("font-weight", "bold");
+      d3.select(this).select("text").style("text-decoration", "underline");
+    };
+
+    function dblClick(d) {
+      if (d3.event.defaultPrevented) return; // click suppressed
+
+      if (d._children || d.children) {
+        toggleChildren(d);
+        update(d);
+        centerNode(d);
+      }
+    };
+
+    function update(source) {
+      // Compute the new height, function counts total children of root node
+      // and sets tree height accordingly. This prevents the layout looking
+      // squashed when new nodes are made visible or looking sparse when nodes
+      // are removed. This makes the layout more consistent.
+      var levelWidth = [1],
+
+      childCount = function(level, n) {
+        if (n.children && n.children.length > 0) {
+          if (levelWidth.length <= level + 1) levelWidth.push(0);
+
+          levelWidth[level + 1] += n.children.length;
+          n.children.forEach(function(d) {
+            childCount(level + 1, d);
+          });
         }
+      },
 
-        // Shows or hides children depending on the current state.
-        function toggleChildren(d) {
-            if (d.children) {
-                d._children = d.children;
-                d.children = null;
-            } else if (d._children) {
-                d.children = d._children;
-                d._children = null;
-            }
-
-            return d;
-        }
-
-        // Toggle children on click.
-
-        function click(d) {
-            if (d3.event.defaultPrevented) return; // click suppressed
-
-            toggleSelectedNode.call(this);
-            // Load chart.
-            var values    = d.load,
-                loadChart = window.LoadChart;
-
-            var chartData = [{
-                key:    d.name,
-                area:   true,
-                values: values
-            }]
+      newHeight, nodes, links, nodeEnter, nodeUpdate, nodeExit, link;
 
-            if (d.capacity) {
-                chartData.push({
-                    key: 'Capacity',
-                    color: 'darkred',
-                    values: values.map(function(sample) {
-                        return ({ x: sample.x, y: d.capacity });
-                    })
-                });
-            }
+      childCount(0, root);
+      newHeight = d3.max(levelWidth) * 35; // 25 pixels per line
 
-            $('.load-graph').empty().append('<svg></svg>');
+      // Update the SVG height to fit the contents.
+      var svgDivHeight = Math.min(newHeight, maxViewerHeight);
+      var baseSvgSelector = $(baseSvg[0][0]);
+      baseSvgSelector.height(newHeight + 50);
+      baseSvgSelector.parent().height(svgDivHeight + 50);
```

Comment:
@antw  these are the only actually 'changed' lines. The other changes are all due to re-indentation.